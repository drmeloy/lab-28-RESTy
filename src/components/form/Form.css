.Form {
  grid-area: form;
  display: grid;
  grid-template-columns: 2fr 1fr;
  grid-template-rows: 40px 40px 1fr;
  grid-template-areas: 
    "url url"
    "methods methods"
    "reqbody auth";
  border: 1px solid;
  margin: 10px;
  padding: 10px;
  background: #ffffff4a;
  border-radius: 25px;
  font-family: 'Oswald', sans-serif;

  textarea:disabled {
    background: #bbb9b9;
  }
}

.url{
  grid-area: url;

  input {
    width: 90%;
    border-radius: 10px;
    padding-left: 10px;
    font-size: .75rem;
    background: #ffffffa8;
    font-family: 'Oswald', sans-serif;
  }
}

.methods {
  grid-area: methods;
  
  input[type='radio'] {
    opacity: 0;
    position: fixed;
    width: 0;
  }

  label {
    display: inline-block;
    background-image: url('../../assets//trianglify-lowres\ \(4\).png');
    padding: 10px 20px;
    font-family: 'Oswald', sans-serif;
    font-size: 16px;
    border: 1px solid #FF5722;
    border-radius: 5px;
    margin-right: 10px;
    cursor: pointer;
  }

  input[type='radio']:checked + label {
    background-image: url('../../assets//trianglify-lowres\ \(5\).png');
    color: white;
    border: 1px solid white;
  }

  button {
    background-image: url('../../assets/trianglify-lowres\ \(6\).png');
    height: 34px;
    width: 90px;
    border: 1px solid #07d507;
    border-radius: 5px;
    margin-right: 10px;
    cursor: pointer;
  }

  button:active {
    background-image: url('../../assets/trianglify-lowres\ \(7\).png');
    color: white;
    border: 1px solid white;
  }
}

.reqbody {
  grid-area: reqbody;
  margin: 10px 10px 0 0;
  padding: 5px;
  border-radius: 12px;
  background: #ffffffa8;
  font-family: 'Oswald', sans-serif;
}

.auth {
  grid-area: auth;

  input {
    margin: 0 10px 10px 0;
    border-radius: 5px;
    font-size: .75rem;
    background: #ffffffa8;
    font-family: 'Oswald', sans-serif;
    padding-left: 5px;
  }

  p {
    margin-bottom: 5px;
  }
}