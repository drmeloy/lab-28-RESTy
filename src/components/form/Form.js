import React from 'react';
import PropTypes from 'prop-types';
import styles from './Form.css';

const Form = ({ url, method, requestBody, username, password, token, onChange, onSubmit, clearForm }) => {
  return (
    <form className={styles.Form} onSubmit={onSubmit}>
      <div className={styles.url}>
        <input required type='text' name='url' placeholder='URL' value={url} onChange={onChange}></input>
      </div>
      <div className={styles.methods}>
        <input className='method-selector' id={'GET'}type='radio' name='method' value='GET' checked={method === 'GET'} onChange={onChange}></input>
        <label htmlFor='GET'>GET</label>
        <input className='method-selector' id='POST' type='radio' name='method' value='POST' checked={method === 'POST'} onChange={onChange}></input>
        <label htmlFor='POST'>POST</label>
        <input className='method-selector' id='PUT' type='radio' name='method' value='PUT' checked={method === 'PUT'} onChange={onChange}></input>
        <label htmlFor='PUT'>PUT</label>
        <input className='method-selector' id='PATCH' type='radio' name='method' value='PATCH' checked={method === 'PATCH'} onChange={onChange}></input>
        <label htmlFor='PATCH'>PATCH</label>
        <input className='method-selector' id='DELETE' type='radio' name='method' value='DELETE' checked={method === 'DELETE'} onChange={onChange}></input>
        <label htmlFor='DELETE'>DELETE</label>
        <button className='form-submit' type='submit'>Go!</button>
        <button className='clear-button' type='reset' onClick={clearForm}>Clear All</button>
      </div>
      <textarea disabled={method === 'GET' || method === 'DELETE'} className={styles.reqbody} placeholder='Raw JSON Body' name='requestBody' value={requestBody} onChange={onChange}></textarea>
      <div className={styles.auth}>
        <p>Basic Authorization</p>
        <input className='username' type='text' name='username' placeholder='Username' value={username} onChange={onChange}></input>
        <input className='password' type='password' name='password' placeholder='Password' value={password} onChange={onChange}></input>
        <p>Bearer Token</p>
        <input className='text' type='text' name='token' placeholder='Bearer Token' value={token} onChange={onChange}></input>
      </div>
    </form>
  );
};

Form.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  requestBody: PropTypes.string,
  username: PropTypes.string,
  password: PropTypes.string,
  token: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Form;
