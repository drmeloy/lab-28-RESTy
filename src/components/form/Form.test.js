import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form url={''} method={''} requestBody={''} username={''} password={''} token={''}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
