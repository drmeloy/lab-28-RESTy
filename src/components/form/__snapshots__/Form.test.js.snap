// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  className="Form"
>
  <div
    className="url"
  >
    <input
      name="url"
      placeholder="URL"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="methods"
  >
    <input
      checked={false}
      className="method-selector"
      id="GET"
      name="method"
      type="radio"
      value="GET"
    />
    <label
      htmlFor="GET"
    >
      GET
    </label>
    <input
      checked={false}
      className="method-selector"
      id="POST"
      name="method"
      type="radio"
      value="POST"
    />
    <label
      htmlFor="POST"
    >
      POST
    </label>
    <input
      checked={false}
      className="method-selector"
      id="PUT"
      name="method"
      type="radio"
      value="PUT"
    />
    <label
      htmlFor="PUT"
    >
      PUT
    </label>
    <input
      checked={false}
      className="method-selector"
      id="PATCH"
      name="method"
      type="radio"
      value="PATCH"
    />
    <label
      htmlFor="PATCH"
    >
      PATCH
    </label>
    <input
      checked={false}
      className="method-selector"
      id="DELETE"
      name="method"
      type="radio"
      value="DELETE"
    />
    <label
      htmlFor="DELETE"
    >
      DELETE
    </label>
    <button
      className="form-submit"
      type="submit"
    >
      Go!
    </button>
    <button
      className="clear-button"
      type="reset"
    >
      Clear All
    </button>
  </div>
  <textarea
    className="reqbody"
    disabled={false}
    name="requestBody"
    placeholder="Raw JSON Body"
    value=""
  />
  <div
    className="auth"
  >
    <p>
      Basic Authorization
    </p>
    <input
      className="username"
      name="username"
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      className="password"
      name="password"
      placeholder="Password"
      type="password"
      value=""
    />
    <p>
      Bearer Token
    </p>
    <input
      className="text"
      name="token"
      placeholder="Bearer Token"
      type="text"
      value=""
    />
  </div>
</form>
`;
